typing/typedecl.ml

```diff
@@ -1839,10 +1839,7 @@ let update_decl_jkind env dpath decl =
       assert (not (Jkind.is_best decl.type_jkind));
       decl, decl.type_jkind
     | Type_open ->
-      let type_jkind =
-        Jkind.Builtin.value ~why:Extensible_variant
-        |> Jkind.mark_best
-      in
+      let type_jkind = Jkind.Builtin.value ~why:Extensible_variant in
```

Comment:
I agree with not marking the other one... but this one I think is important. It allows, I think, the following:

```
type open_ = ..
type t : immediate with open_

[%%expect {|
type open_ = ..
type t
|}]
```

where we normalize away the kind of `t` to just be `value`.

I know it's awkward because we can't test this in this patch, so I would either restore the `mark_best` here or leave a comment saying to do so in the later patch (where we can test and verify my suspicion).

Really, maybe what I'm saying is that the quality.hs test in the future patch should have tests with extensible types, which I think were left out. I will make a comment saying so.